### PR TITLE
Bump web compilers

### PR DESCRIPTION
**Pull request by DanTup**
@jakemac53 @jacob314  This PR bumps the build_web_compilers version, which breaks the build. I found that there are JavaScript errors occurring that were being silently swallowed, so I've also:

1. Wrapped analytics setup in try/catch so we never stop the app loading for analytics failures
2. Logged any other errors to console.error so they're not silent

In the new results, there are many failures - all which seem related to things using external JavaScript:

> exception: {type: object, subtype: error, className: TypeError, description: TypeError: src__ui__analytics.isGtagsEnabled is not a function

And even stranger:

```
chrome • console:log • DevTools version 0.1.0.
chrome • console:error • null
chrome • console:error • null
chrome • console:error • null
chrome • console:error • null
```

I'm not sure what's going on there 🤷‍♂️

I don't know if these are issues in DevTools, or might be breaking changes in `build_web_compilers`. Any ideas?

### CODE REVIEW

**Comment by jakemac53** [comment]
Most likely the issues reside in the dev_compiler itself (specifically, the new kernel backend for it). I would file an SDK issue.

**Comment by jacob314** [comment]
I'll take a look. Likely it is a compiler inconsistency with JS interop that we can workaround for now.

**Comment by DanTup** [comment]
@jacob314 ok, I'll let you file an SDK issue then since you'll no doubt have a better understanding of it after looking into it :-)

**Comment by DanTup** [comment]
Sounds like https://github.com/dart-lang/sdk/issues/36912 is the issue here. I presume once that's fixed, we'll have to bump the minimum version to start from the version with the fix (since we can't do a range with an exclusion in the middle?)

**Comment by jakemac53** [comment]
Correct, sorry I meant to link this issue as well when I filed that :).

**Comment by devoncarew** [comment]
@DanTup - this may not be needed anymore? Are there parts of this PR we should still land?

**Comment by DanTup** [comment]
Yeah, there's a fix for bad string interpolation that outputs `.toString()` and also handling of exceptions for analytics (without this, analytics failing to load stops the whole app from loading). I'm not sure why I didn't put them in another PR though, but I can do.

Has the other JS issue been fixed? Do we not still want to bump the webdev version?

**Comment by devoncarew** [comment]
I believe we do work with the latest webdev; https://github.com/flutter/devtools/pull/702.

**Comment by DanTup** [comment]
Whoops, when I set webdev I meant build_web_compilers. There was an issue with interop (https://github.com/dart-lang/sdk/issues/36912). I think it may be fixed (in 2.3.2-dev.0.1?) based on [this comment](https://github.com/dart-lang/sdk/issues/36944#issuecomment-499179141).

I'll rebase this on latest so the CI runs again and we can see where it is. If it fails, I'll split the other changes out.

**Comment by DanTup** [comment]
Looks like this works on dev builds for DDC now, though (expectedly) fails on stable. We could temporarily skip stable DDC builds until the next stable Dart release, or land this after that?

**Comment by DanTup** [comment]
I've moved the non-web_compilers change out to #722 so they can be landed more easily. This PR is now just bumping the compilers, which works for dev but not for stable. I *think* we're save to land this once we're happy that everyone developing on/publishing devtools is using an SDK that has the fix.

**Comment by DanTup** [comment]
I've pushed a change that disables the DDC bots for Stable temporarily so this should go green if we want to land it (and re-enable them after the next Dart stable release).

File: .travis.yml [hunk 1]
```diff
@@ -17,6 +17,10 @@ matrix:
   #     we can build on an older SDK.
   - dart: stable
     env: BOT=main
+  # TODO(dantup): Temporarily skip stable DDC builds until the interop fix is
```
Comment (jacob314) [inline .travis.yml]: add a bug # for this so we get reminded to uncomment the tests.

Comment (DanTup) [reply .travis.yml]: I've opened https://github.com/flutter/devtools/issues/748 and linked it back here.

**Pull request merged**